### PR TITLE
fix: more audit remediations 

### PR DESCRIPTION
**Pull request by shaspitz**
## Describe your changes

More audit fixes! These ones don't necessarily correspond to findings from the official report, as some of them are low/info severity. 

Changes:

* `ValidatorOptInRouter.areValidatorsOptedIn` now returns a struct containing three booleans representing which registries a validator is registered with. This will be useful for oracle slashing logic. Commit: https://github.com/primev/mev-commit/pull/470/commits/6d49ffdf32ad6c5c4e3d850677729932866a0868. This PR includes off-chain updates for this change in api. 
* Setter for `Gateway.relayer` was added, along w/ missing tests for related setters. Commit: https://github.com/primev/mev-commit/pull/470/commits/530add7ee332632b4ea48d5ddfcb6138efa1e86c
* We now check the return value of `IEnumerableSet.add` in the middleware contract, and revert if the `add` fails. Note this revert shouldn't happen and would indicate state corruption. Commit: https://github.com/primev/mev-commit/pull/470/commits/9c39370f8dc26664c74eec5522990ff246cf6857
* Adds documentation on how the oracle should handle frozen validators registering with other registries. Commit: https://github.com/primev/mev-commit/pull/470/commits/f9a200cebe4e344b5759070a14b19b458d003050
* Adds `wouldVaultBeValidWith` function to the middleware contract, along with documentation. The new function assists the owner account in confirming preconditions prior to calling `setSlashPeriodSeconds`. Commit: https://github.com/primev/mev-commit/pull/470/commits/42237606d62580c9490073150588bdc24a85d231
* Updates our eigenlayer reference to their latest [mainnet commit](https://github.com/Layr-Labs/eigenlayer-contracts/tree/mainnet), and updates our mock implementations for their contracts. Commit: https://github.com/primev/mev-commit/pull/470/commits/c7b4e877516cedc1bf0c415692cbb184464b2366
* Updates symbiotic reference to their latest main commit, note they don't have a specified _production version_ to reference yet. Commit: https://github.com/primev/mev-commit/pull/470/commits/5c71ecedee72415fd0aa8f116b99bc287c177465
* Removes any modifiers from MevCommitAVS that could possibly change in state during the underlying function call, following solidity best practice. Commit: https://github.com/primev/mev-commit/pull/470/commits/e6155f42cd70a55e12f5901364526f94c2bdbc1d. 
* Further simplify symbiotic veto slashing logic. This changes the symbiotic middleware slashing protocol such that the oracle calls requestSlash and executeSlash in the same tx. This builds off https://github.com/primev/mev-commit/pull/466. Commit: https://github.com/primev/mev-commit/pull/470/commits/d1f7fdd9b2c49ecc97b0fc1407fdc1d0732af032.

## Checklist before requesting a review

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by shaspitz** [comment]
ci failing seems unrelated to these changes, see https://github.com/primev/mev-commit/pull/471

**Comment by shaspitz** [comment]
ci failing is due to outage: https://status.canonical.com/, will re-run tomorrow

File: contracts/contracts/validator-registry/avs/MevCommitAVS.sol [hunk 1]
```diff
@@ -204,21 +165,31 @@ contract MevCommitAVS is IMevCommitAVS, MevCommitAVSStorage,
     /// @dev Allows a validator to request deregistration from the MevCommitAVS.
     /// @notice For each validator the underlying _requestValidatorDeregistration enforces the sender is either
     /// the podOwner, delegated operator, or the contract owner.
-    function requestValidatorsDeregistration(bytes[] calldata valPubKeys)
-        external whenNotPaused() onlyRegisteredValidators(valPubKeys) onlyPodOwnerOrOperatorOfValidators(valPubKeys) {
+    function requestValidatorsDeregistration(bytes[] calldata valPubKeys) external whenNotPaused() {
         uint256 len = valPubKeys.length;
         for (uint256 i = 0; i < len; ++i) {
```
Comment (Mikelle) [inline contracts/contracts/validator-registry/avs/MevCommitAVS.sol:170]: nit to reduce storage reads:
`bytes valPubKey = valPubKeys[i];`
and use that value and same approach later on

Comment (shaspitz) [reply contracts/contracts/validator-registry/avs/MevCommitAVS.sol:170]: Fixed in https://github.com/primev/mev-commit/commit/013cc5029a2b0ab567d388084e7bff5c7d443a55

**Review by Mikelle — APPROVED** [review]

**Pull request merged**